base/audit_events.go

```diff
@@ -64,17 +65,20 @@ const (
 	AuditIDReadDatabaseConfig   AuditID = 54010
 	AuditIDUpdateDatabaseConfig AuditID = 54011
 	// Database operation events
-	AuditIDDatabaseOffline       AuditID = 54020
-	AuditIDDatabaseOnline        AuditID = 54021
-	AuditIDDatabaseCompactStatus AuditID = 54030
-	AuditIDDatabaseCompactStart  AuditID = 54031
-	AuditIDDatabaseCompactStop   AuditID = 54032
-	AuditIDDatabaseResyncStatus  AuditID = 54040
-	AuditIDDatabaseResyncStart   AuditID = 54041
-	AuditIDDatabaseResyncStop    AuditID = 54042
-	AuditIDDatabasePostUpgrade   AuditID = 54043
-	AuditIDDatabaseRepair        AuditID = 54044
-	AuditIDDatabaseFlush         AuditID = 54045
+	AuditIDDatabaseOffline                 AuditID = 54020
+	AuditIDDatabaseOnline                  AuditID = 54021
+	AuditIDDatabaseAttachmentCompactStatus AuditID = 54030
+	AuditIDDatabaseAttachmentCompactStart  AuditID = 54031
+	AuditIDDatabaseAttachmentCompactStop   AuditID = 54032
+	AuditIDDatabaseTombstoneCompactStatus  AuditID = 54033
+	AuditIDDatabaseTombstoneCompactStart   AuditID = 54034
+	AuditIDDatabaseTombstoneCompactStop    AuditID = 54035
```

Comment:
I waffled on this, because it's easy to drop fields if they aren't mandatory and this was the way it was written originally. I don't think any of these parameters are super important though.

The sticking point to me was that tombstone is necessarily query based and doesn't support checkpoints but attachment compaction does support resume.

Do you have feelings on `type` as a generic field, or my original idea of `compaction_type`? I worry about what to name this constant and the likelihood we might accidentally use `type` to mean something else.

We had a near mishap with overlapping field names before: `method` referred to `auth_method` (basic/oidc) and `http_method` (GET/POST/PUT). The implementation as is will take fields from `Audit(ctx, id, fields)` and have the context fields overwrite the passed in fields. I do not think there's ever a reason we'd implement this.

That said, I am not sure it's a big deal to have `type` mean different things in different contexts.

An alternate way of handling this would be to log query string for this event type, like we do for the payload.

Reply:
Decided to go with separating fields from offline conversation.